core/src/test/scala/unit/kafka/integration/KafkaServerTestHarness.scala

```diff
@@ -67,6 +68,14 @@ abstract class KafkaServerTestHarness extends QuorumTestHarness {
    */
   def generateConfigs: Seq[KafkaConfig]
 
+  /**
+   * It is sometimes useful to keep the same log.dirs configuration value; override this method if that is desired
+   *
+   * @param priorConfigs the prior configs
+   * @return the new generated configs
+   */
+  def regenerateConfigs(priorConfigs: Seq[KafkaConfig]): Seq[KafkaConfig] = generateConfigs
+
```

Comment:
The other option is to change the existing `generateConfigs()` method to take a `priorConfigs: Option[Seq[KafkaConfig]]` argument.  This is a more intrusive change as it would require 30+ changes elsewhere, but I am not averse to doing it if that is preferred.

Reply:
Isn't there a simpler way of doing this where subclasses who want the old configs just do `super.generateConfigs` and then modify what that function returns? I'm not convinced we need a new function or argument here.

Reply:
> a simpler way of doing this where subclasses who want the old configs just do super.generateConfigs and then modify what that function returns

The `generateConfigs()` method is abstract, so `super.generateConfigs` isn't there.  The real issue here is that the class generally doesn't specify it's log directory -- that's typically auto-generated.  So if we generate configs a second time we get a new log directory, and typically we don't want a new one -- we want to keep the old one.  So we need the prior generated configs in order to know that.

Reply:
I was able to remove this new `regenerateConfigs()` method and just implement some logic in the `ServerShutdownTest` class to save away the prior configs and react accordingly.  I'm fine with doing it this way given that reconfiguration is rarely used and this solution is pretty easy to implement in test classes.